### PR TITLE
Acyclic types are closed under equivalences

### PR DESCRIPTION
**Pull request by tomdjong**
Relies on suspensions preserving equivalences (and retracts)

### CODE REVIEW

**Comment by tomdjong** [comment]
Naming suggestions/corrections are appreciated!

File: src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md [hunk 1]
```diff
@@ -190,3 +194,65 @@ module _
...
+  pr2 (suspension-retract-of-suspension-retract-of R) =
+    retraction-map-suspension-retraction
+      ( section-retract-of R)
+      ( retraction-section-retract-of R)
+```
+
+### Equivalent types have equivalent suspensions
+
+```agda
+  is-equiv-map-suspension-is-equiv :
```
Comment (VojtechStep) [inline src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md:246]: We prefer not to have modules open across headings - could you open a new anonymous module for these two definitions?

**Comment by tomdjong** [comment]
@fredrik-bakke I can't add the label myself, can I?

**Comment by fredrik-bakke** [comment]
No, don't think so. I wonder if we can enable that...

**Comment by fredrik-bakke** [comment]
I'm writing a file on retracts of maps at the moment, btw. Then you can show that acyclic maps are closed under retracts as well :)

**Comment by fredrik-bakke** [comment]
> No, don't think so. I wonder if we can enable that...

Doesn't seem like there's a simple solution to this. But don't worry about it, I like labeling things :)

File: src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md [hunk 2]
```diff
@@ -190,3 +194,69 @@ module _
...
+    homotopy-reasoning
+      map-suspension r ∘ map-suspension f
+      ~ map-suspension (r ∘ f)
+        by inv-preserves-comp-map-suspension f r
+      ~ map-suspension id
+        by htpy-eq (ap map-suspension (eq-htpy h))
+      ~ id
+        by id-map-suspension A
+
+  suspension-retract-of-suspension-retract-of :
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md:233]: The following is probably more consistent
```agda
retract-of-suspension-retract-of
```

File: src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md [hunk 3]
```diff
@@ -190,3 +194,69 @@ module _
...
+      ~ map-suspension id
+        by htpy-eq (ap map-suspension (eq-htpy h))
+      ~ id
+        by id-map-suspension B
+
+  retraction-map-suspension-retraction :
+    (f : A → B) → retraction f → retraction (map-suspension f)
+  pr1 (retraction-map-suspension-retraction f S) =
+    map-suspension (map-retraction f S)
+  pr2 (retraction-map-suspension-retraction f (r , h)) =
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md:223]: These names are good. I've also been trying out `has-retraction-map-suspension-has-retraction`, to make it clear which direction the retraction goes, so you can consider if you prefer that.

File: src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md [hunk 4]
```diff
@@ -190,3 +194,69 @@ module _
...
 ```
+
+### Suspensions preserve retracts
+
+```agda
+module _
+  {l1 l2 : Level} {A : UU l1} {B : UU l2}
+  where
+
+  section-map-suspension-section :
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md:205]: Another option is `has-section-map-suspension-has-section`

File: src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md (hunk 3 above)
Comment (tomdjong) [reply src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md]: I like that :slightly_smiling_face:

File: src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md [hunk 5]
```diff
@@ -202,11 +202,11 @@ module _
   {l1 l2 : Level} {A : UU l1} {B : UU l2}
   where
 
-  section-map-suspension-section :
+  has-section-map-suspension-has-section :
     (f : A → B) → section f → section (map-suspension f)
```
[the remark this answers is not in the document]
Comment (tomdjong) [reply src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md]: Fair enough! I've reverted the changes.

[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md]: Maybe we _should_ rename `section` to `has-section`

[the remark this answers is not in the document]
Comment (tomdjong) [reply src/synthetic-homotopy-theory/functoriality-suspensions.lagda.md]: @EgbertRijke Are you happy with the other names? This discussion has moved to Discord now anyway.

**Comment by fredrik-bakke** [comment]
I'll leave this PR open until Egbert's had a chance to respond. Apologies if I've caused any confusion, and thanks again for the contribution, Tom.

**Comment by fredrik-bakke** [comment]
I'll merge this PR now to not hold you up further; I think @EgbertRijke will agree with your current namings.

**Pull request merged**